app/src/main/java/fr/neamar/kiss/CustomIconDialog.java

```diff
@@ -0,0 +1,455 @@
+                mSearch.post(() -> refreshList());
+            }
+        });
+
+        Bundle args = getArguments() != null ? getArguments() : new Bundle();
+        @Nullable
+        ComponentName cn = ComponentName.unflattenFromString(args.getString("className", ""));
+        UserHandle userHandle = args.getParcelable("userHandle");
+        String name = args.getString("componentName", "");
+        long customIcon = args.getLong("customIcon", 0);
+
+        IconsHandler iconsHandler = KissApplication.getApplication(context).getIconsHandler();
+
+        // Preview
+        {
+            mPreview = view.findViewById(R.id.preview);
+            Drawable drawable = customIcon != 0 ? iconsHandler.getCustomIcon(name, customIcon) : null;
+            if (drawable == null)
+                drawable = iconsHandler.getDrawableIconForPackage(cn, userHandle);
+            mPreview.setImageDrawable(drawable);
+        }
+
+        // OK button
+        {
+            View button = view.findViewById(android.R.id.button1);
+            button.setOnClickListener(v -> {
+                if (mOnConfirmListener != null) {
+                    mOnConfirmListener.onConfirm(mSelectedDrawable);
+                }
+                dismiss();
+            });
+        }
+
+        // CANCEL button
+        {
+            View button = view.findViewById(android.R.id.button2);
+            button.setOnClickListener(v -> dismiss());
+        }
+
+        ViewGroup quickList = view.findViewById(R.id.quickList);
+
+        // add default icon
+        {
+            Drawable drawable = iconsHandler.getDrawableIconForPackage(cn, userHandle);
+
+            ImageView icon = quickList.findViewById(android.R.id.icon);
+            icon.setImageDrawable(drawable);
+            icon.setOnClickListener(v -> {
+                mSelectedDrawable = null;
+                mPreview.setImageDrawable(((ImageView) v).getDrawable());
+            });
+            ((TextView) quickList.findViewById(android.R.id.text1)).setText(R.string.default_icon);
+        }
+
+        IconPack iconPack = iconsHandler.getCurrentIconPack();
+        //IIconPack systemIconPack = iconsHandler.getSystemIconPack();
+
+        // add getActivityIcon(componentName)
+        {
+            Drawable drawable = null;
+            try {
+                drawable = context.getPackageManager().getActivityIcon(cn);
+            } catch (PackageManager.NameNotFoundException ignored) {
+            }
+            if (drawable != null) {
+                addQuickOption(R.string.custom_icon_activity, drawable, quickList);
+                if (iconPack != null)
+                    addQuickOption(R.string.custom_icon_activity_with_pack, iconPack.applyBackgroundAndMask(context, drawable), quickList);
+//                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O)
+//                    addQuickOption(R.string.custom_icon_activity_adaptive, DrawableUtils.applyIconMaskShape(context, drawable), quickList);
```

Comment:
Commented code here. What was this for?

Reply:
After we have the PR #1434 merged this will add another option for custom icons